### PR TITLE
Credentials are now passed to STS client correctly on validation. Bad…

### PR DESCRIPTION
**Pull request by bryceitoc9**
… credentials pop a toast on init. Callers are responsible for cleaning up credentials

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description

More fixes and improvements for credential validation.
* Credentials are now sent to the STS client correctly; previously, changing accounts initially validated with the old profile's credentials.
* Responsibility for removing bad profiles is moved to the caller
* The user is notified via toast that a profile is bad when a user is logged out.

## Motivation and Context

Fixing a bug and making the code base more maintainable.

## Related Issue(s)

N/A

## Testing

More exhaustive slate of tests in addition to the unit tests:
* User activates toolkit and uses existing credentials profile
* User activates toolkit and loads a valid profile
* User activates toolkit and loads an invalid profile — bad credentials
* User activates toolkit and loads an invalid profile — profile doesn't exist
* User changes valid active profile to another valid profile
* User changes valid active profile to invalid profile — bad credentials
* User logs out and logs back in — good credentials
* User logs out and logs back in — bad credentials

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/defaultAwsContextCommands.ts [hunk 1]
```diff
@@ -87,13 +87,13 @@ export class DefaultAWSContextCommands {
...
+                // tslint:disable-next-line: no-floating-promises
+                UserCredentialsUtils.badUserCredentialPrompt(profileName)
             }
         }
-        if (!profileName || !successfulLogin) {
-            // credentials are invalid. Prompt user and log out
-            // TODO: Prompt user!
-            await this.onCommandLogout()
-        }
+        // do nothing if !profileName: that means user didn't select from the quick pick.
```
Comment (awschristou) [inline src/shared/defaultAwsContextCommands.ts]: If we explicitly exit after line 83, the rest of the function is dedicated to dealing with logging in

File: package.nls.json [hunk 2]
```diff
@@ -62,6 +62,7 @@
     "AWS.log.invalidLevel": "Invalid log level: {0}",
     "AWS.message.loading": "Loading...",
     "AWS.message.credentials.error": "There was an issue trying to use credentials profile {0}: {1}",
+    "AWS.message.credentials.invalidprofile": "Credentials profile {0} is invalid",
```
Comment (mpiroc) [inline package.nls.json]: Should we include a link to the credentials profile documentation?

File: src/shared/credentials/userCredentialsUtils.ts [hunk 3]
```diff
@@ -194,12 +199,8 @@ export class UserCredentialsUtils {
                 return true
             }
         } catch (err) {
-            // swallow any errors--anything that isn't a success should end in the same state:
-            // no set profile + return false
+            // swallow any errors--anything that isn't a success be handled as a failure by the caller
```
Comment (mpiroc) [inline src/shared/credentials/userCredentialsUtils.ts]: Nit: Missing a word ("*should* be handled"?)

File: src/shared/credentials/userCredentialsUtils.ts [hunk 4]
```diff
@@ -213,4 +214,14 @@ export class UserCredentialsUtils {
         await awsContext.setCredentialProfileName()
         await awsContext.setCredentialAccountId()
     }
+
+    public static async badUserCredentialPrompt(profileName: string) {
```
Comment (mpiroc) [inline src/shared/credentials/userCredentialsUtils.ts]: Nit: Function names should be actions, i.e. `showBadUserCredentialPrompt` or `promptUserToFixCredentials`.

File: src/shared/defaultAwsContextCommands.ts (hunk 1 above)
Comment (bryceitoc9) [reply src/shared/defaultAwsContextCommands.ts]: ✔️

File: src/shared/credentials/userCredentialsUtils.ts (hunk 3 above)
Comment (bryceitoc9) [reply src/shared/credentials/userCredentialsUtils.ts]: ✔️

File: src/shared/credentials/userCredentialsUtils.ts (hunk 4 above)
Comment (bryceitoc9) [reply src/shared/credentials/userCredentialsUtils.ts]: ✔️

File: src/shared/defaultAwsContextCommands.ts [hunk 5]
```diff
@@ -81,19 +81,19 @@ export class DefaultAWSContextCommands {
...
+            return
+        }
+        const successfulLogin = await UserCredentialsUtils.addUserDataToContext(profileName, this._awsContext)
+        if (successfulLogin) {
+            this.refresh()
+            await this.checkExplorerForDefaultRegion(profileName)
+        } else {
+            await this.onCommandLogout()
+            // tslint:disable-next-line: no-floating-promises
+            UserCredentialsUtils.notifyUserCredentialsAreBad(profileName)
```
Comment (mpiroc) [inline src/shared/defaultAwsContextCommands.ts]: Nit: It's fine to await here, because `onCommandLogin` is already fire-and-forget. Same comment below.

**Pull request merged**